src/plugins/intel_npu/src/compiler/src/graph_transformations.cpp

```diff
@@ -6,29 +6,55 @@
 
 #include <istream>
 #include <mutex>
+#include <streambuf>
 
 #include "openvino/pass/serialize.hpp"
 #include "transformations/op_conversions/convert_interpolate11_downgrade.hpp"
 
 namespace intel_npu::driverCompilerAdapter {
 
-IR serializeToIR(const std::shared_ptr<const ov::Model>& origModel, uint32_t supportedOpset) {
+IR::IR(const std::shared_ptr<const ov::Model>& origModel, uint32_t supportedOpset, bool largeModel)
+    : _logger("LevelZeroCompilerAdapter::IR", Logger::global().level()),
+      _isLargeModel(largeModel),
+      _xmlStream(&_xmlCache),
+      _weightsStream(&_weightsCache) {
```

Comment:
I'm wondering if there's any way we can find out the size of the graph/weights in advance in order to allow us to pre-allocate a reasonable size for these buffers. This should help the performance a bit, right? Perhaps `ov::Model::get_graph_size()` could be useful?

I won't insist in doing this research right now, we may optimize this later if we decide to do so. For now I'm just looking for some feedback, and we may eventually open a ticket after that.

Reply:
There is room to reserve some memory in streambuffer to do optimization, but get_graph_size() only give us whole model size, we can not get size of xml and weights now. For large model, weights is GB level but xml is MB level, may need to find some experience value for this later, feel free to create ticket for this, we will also compare this path with memory map path.

Reply:
We could consider deque (at least for weights), see other comment.

Reply:
Tried deque but get INVALID_POINTER_WRITE_c0000005_VCRUNTIME140.dll!memcpy on Windows, The reason is basic_streambuf and iostream are using pointer to mange memory inside code and require continuous memory region. Vector guarantee the memory to be continuous and provide data() API but deque may use memory blocks. So will not use deque since we do not know how other platform and release deal with the structure.

Reply:
streambuf must maintain a "buffered portion" of the underlying character sequence,
that portion must be contiguous, but for weights we most likely will get just `read` calls where the stream just calls xsgetn directly, for get/getline etc. stream will use the buffered portion via exposed pointers (probably mostly used for XML).
We can implement that buffered portion by coping some bytes from deque into a vector just for sake of satisfying the stream contract.
For weight most work will likely done by xsgetn/xsputn so will not use that intermediate buffer.